test/test-parse-statement.cpp

```diff
@@ -1244,6 +1244,61 @@ TEST_F(Test_Parse_Statement, disallow_label_named_yield_in_generator_function) {
                             "visit_exit_function_scope",
                         }));
 }
+
+TEST_F(Test_Parse_Statement, if_body_with_semicolon_typescript) {
```

Comment:
We should have a similar test but for `else`'s body. E.g. `if (a) {} else;`

Reply:
(According to [your comment](https://github.com/quick-lint/quick-lint-js/issues/1067#issuecomment-1712256942), such code should produce no diagnostic.)